### DISCUSSION TITLE
NuxtLink takes to a different url than its to="(...)" value

### DISCUSSION THREAD

**Discussion by DamianGlowala** [nuxt/nuxt#17433] [Questions]
I stumbled upon a really weird behavior with `<NuxtLink>`.

In my project, the page available at `/admin-dashboard` doesn't have a template itself; it only contains a middleware which redirects to the first one of the other pages available within the admin dashboard (e.g. `/admin-dashboard/accounts`).

However, one of the `<NuxtLink>`'s on the website leading to `/admin-dashboard`, actually leads to an address I have entered a long time ago (e.g. `/admin-dashboard/<old-path>`). The issue is, that the value of `<old-path>` has been removed/changed and does not even exist in the codebase. Is there any caching system which keeps the old url and redirects to a page which no longer exists, and therefore leads to 404?

When `/admin-dashboard` is accessed directly from the browser, the middleware does redirect correctly to e.g. `/admin-dashboard/accounts`.
However, when accessed through the `<NuxtLink>`, it redirects to a no longer existing url.

Help on the above matter would be highly appreciated!

A very _confused_ Nuxter :)

**Comment by Flanker72** [comment]
This may be related to `vite` bundler cache. When you are building/running your project, Nuxt products `.nuxt` (for dev) and `.output` (for prod) folders. They contain all builded data chunks. If somehow the old chunk is stored there, it will be delivered to the browser, providing wrong content. You may try to delete generated .nuxt/.output folders and rebuild the project.

**Reply by DamianGlowala** [reply]
Thanks for the suggestion! I have deleted both folders, run project in the development mode and the same behavior persists. What's odd is that nowhere in my codebase there's a mention of the url that it keeps redirecting to :)

**Reply by Flanker72** [reply]
Please, check the code delivered to your browser. Perform a full-text search within your DevTools and check where this piece of code is located. Also check if you are not enabled any overrides in the DevTools - if some code was modified/debugged locally, it may be cached in the browser. Seems you may have local browser cache issue. Looking for cached results in the network tab may help.

**Reply by DamianGlowala** [reply]
Again, thanks for the useful pointers! It does work in Edge, just checked it :) It must be a browser-specific issue then. Cleared all the cache I could find in the DevTools; inspected the DOM with no luck... Chrome behaves still the same way. Knowing it's not Nuxt, I think I'll live with it for the time being.

**Reply by danielroe** [reply]
I think it's not a browser-specific issue, it's just your browser's _cache_.

**Reply by Flanker72** [reply]
<img width="945" alt="image" src="https://user-images.githubusercontent.com/619699/174069531-a2a1414d-b1cc-4a85-bdf9-c9ecaca0b636.png">

Check this more carefully. If you have any overrides (left panel) or locally saved data (orange exclamation mark) then this data will be used instead of received from the server.
